### PR TITLE
fix bug - gdevelop allows selection of instances on hidden layers

### PR DESCRIPTION
**Pull request by blurymind**
this fixes:
#661 

Implementation after @4ian's review and feedback in 
https://github.com/4ian/GDevelop/pull/669

Closed previous one, because by mistake I mixed it with a testing branch

### CODE REVIEW

File: newIDE/app/src/SceneEditor/index.js [hunk 1]
```diff
@@ -521,7 +521,8 @@ export default class SceneEditor extends React.Component<Props, State> {
   ) => {
     this.instancesSelection.clearSelection();
     instances.forEach(instance =>
-      this.instancesSelection.selectInstance(instance)
+      this.instancesSelection.selectInstance(instance),
```
Comment (4ian) [inline newIDE/app/src/SceneEditor/index.js]: This line is bugged ;)

File: newIDE/app/src/SceneEditor/InstancesSelection.js [hunk 2]
```diff
@@ -24,21 +24,36 @@ export default class InstancesSelection {
...
     if (this.isInstanceSelected(instance)) {
       if (multiselect) this.unselectInstance(instance);
 
       return;
     }
 
     if (!multiselect) this.clearSelection();
-    this.selection.push(instance);
+
+    if (!layersVisibility) {
```
Comment (4ian) [inline newIDE/app/src/SceneEditor/InstancesSelection.js]:
```if (!layersVisibility || layersVisibility[instance.getLayer()]) {
  this.selection.push(instance);
}```

is shorter/clearer :)

**Comment by 4ian** [comment]
Looks like flow typing is failing: it actually catched a bug where you've misplaced a comma ;)
Note that you can run `npm run flow` or `yarn flow` from the app folder on your machine. This will run flow that will report any typing error. Useful here as it caught an error :)

Apart from my comments it should be good then!

File: newIDE/app/src/SceneEditor/index.js (hunk 1 above)
Comment (blurymind) [reply newIDE/app/src/SceneEditor/index.js]: hahaha xD I should really stop coding late after work. It's actually surprising gdevelop didn't crash when I was testing it. That second parameter is no longer optional

**Comment by blurymind** [comment]
@4ian I should really use that flow check more often. It looks like a very valuable tool to catch silly mistakes

**Pull request merged**